code/datums/beam.dm

```diff
@@ -118,6 +118,16 @@
 	anchored = TRUE
 	var/datum/beam/owner
 
+/obj/effect/ebeam/Initialize(mapload)
+	. = ..()
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = PROC_REF(on_atom_entered),
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
+
+/obj/effect/ebeam/proc/on_atom_entered(datum/source, atom/movable/entered)
+	return
```

Comment:
I think we'll probably want to enforce signal handlers on these functions to ensure they don't sleep.